app/models/user.rb

```diff
@@ -119,14 +119,15 @@ class User < ApplicationRecord
   scope :matches_ip, ->(value) { left_joins(:ips).where('user_ips.ip <<= ?', value).group('users.id') }
   scope :emailable, -> { confirmed.enabled.joins(:account).merge(Account.searchable) }
 
-  before_validation :sanitize_languages
   before_validation :sanitize_role
-  before_validation :sanitize_time_zone
-  before_validation :sanitize_locale
   before_create :set_approved
   after_commit :send_pending_devise_notifications
   after_create_commit :trigger_webhooks
 
+  normalizes :locale, with: ->(locale) { I18n.available_locales.exclude?(locale.to_sym) ? nil : locale }
+  normalizes :time_zone, with: ->(time_zone) { ActiveSupport::TimeZone[time_zone].nil? ? nil : time_zone }
+  normalizes :chosen_languages, with: ->(chosen_languages) { chosen_languages.compact_blank!.empty? ? nil : chosen_languages }
```

Comment:
I think the in-place mutation operation (`chosen_languages.compact!`) in the condition makes this line less readable

Reply:
yeah it's a little more awkward than the others for sure ... even the ternary being here at all (in all three) is not the best, IMO. I'll do another round on this and see if I can clean the language up at least, and/or remove it from this PR.

Reply:
Pushed update which removes the ternary checks in favor of intersection checks.